resources/views/tools/database/vue-components/database-column.blade.php

```diff
@@ -0,0 +1,134 @@
+@section('database-column-template')
+
+<tr class="newTableRow">
+    <td>
+        <input v-model.trim="column.name" @input="onColumnNameInput" type="text" class="form-control" required pattern="{{ $table->identifierRegex }}">
+    </td>
+
+    <td>
+        <select :value="column.type" @change="onColumnTypeChange" class="form-control" tabindex="-1">
+        @foreach($table->columnTypes as $category => $types)
+            <optgroup label="{{ $category }}">
+            @foreach($types as $type)
+                <option value="{{ $type }}">{{ strtoupper($type) }}</option>
+            @endforeach
+            </optgroup>
+        @endforeach
+        </select>
+        <div v-if="column.type=='enum'">
+            <small>Enter Values (comma separated)</small>
+            <input type="text" class="form-control" name="enum[]">
+        </div>
+    </td>
+
+    <td>
+        <input v-model.number="column.length" type="number" min="0">
+    </td>
+
+    <td>
+        <input v-model="column.notnull" type="checkbox">
+    </td>
+
+    <td>
+        <input v-model="column.unsigned" type="checkbox">
+    </td>
+
+    <td>
+        <input v-model="column.autoincrement" type="checkbox">
+    </td>
+
+    <td>
+        <select :value="index.type" @change="onIndexTypeChange" :disabled="indexNotSupported" class="form-control" tabindex="-1">
+            <option value=""></option>
+            <option value="INDEX">INDEX</option>
+            <option value="UNIQUE">UNIQUE</option>
+            <option value="PRIMARY">PRIMARY</option>
+        </select>
+    </td>
+
+    <td>
+        <input v-model.trim="column.default" type="text" class="form-control">
+    </td>
+
+    <td>
+        <div class="btn btn-danger delete-row" @click="deleteColumn"><i class="voyager-trash"></i></div>
+    </td>
+
+</tr>
+
+@endsection
+
+<script>
```

Comment:
Maybe we should consider separating this Javascript into its own view to split things up a bit?